contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,81 @@
+
+Here is a checklist for developers who are going through the deprecation process. Although this is a step-by-step guide, it is not a strict process and please use this as a reference.
+
+Note: Component v1 is refered to the component that is going to be deprecated and component v2 is the one that is going to be promoted to the main bundle.
+
+#### Source Code
+
+- [ ] Move the component v1's source code under the `src/deprecated` folder.
+- [ ] Remove the component v1's export in the main bundle to be added to the deprecated one.
+- [ ] Export the component v1 in the deprecated bundle `src/deprecated/index.ts`.
+- [ ] Rename the component v2 to its neutral version if it includes `2` or `v2`. I.e. `ActionList2` -> `ActionList`.
+- [ ] Move the component v2's source code to the `src` folder if it is under `src/drafts`.
+- [ ] Export the component v2 in the main bundle `src/index.ts`.
+
+#### Storybook
+
+- [ ] Update the component v2's storybook title from `Drafts/Components/<Component-Name>` to `Components/<Component-Name>`.
+- [ ] Move the component v1's stories under `src/stories/deprecated`.
+
+#### Tests
+
+- [ ] Move the component v1's tests under `src/__tests__/deprecated`.
+- [ ] Update `script/generate-e2e-tests.js` to reflect the component v2's component ID is generated by the storybook title.
+- [ ] Update the snapshots using the command `npm run test -- -u`.
+
+#### Docs
+
+- [ ] Move the deprecated component's docs to the `src/docs/deprecated` folder and update the title by adding `(legacy)`, status as well as the storybook and the source code link.
+  - [ ] Make sure to update `jsx live` -> `jsx live deprecated`
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react/deprecated"`.
+- [ ] Add a `Deprecation` section with the link of the newly developed component and provide a diff. See [deprecated ActionList docs](https://primer.style/react/deprecated/ActionList#deprecation) as an example.
+- [ ] Move the new components docs from draft folder to the main docs folder `docs/content/` and update the title by removing `v2`, status as well as the storybook and the source code link.
+  - [ ] Make sure to update `jsx live drafts` -> `jsx live`
+  - [ ] Make sure to update the import code block. I.e. `import { ActionList } from "@primer/react"`.
+  - [ ] Update the navigation on `docs/src/@primer/gatsby-theme-doctocat/nav.yml` accordingly.
+
+### Comms side of the deprecation
+
+When Primer team is ready to promote the newly developed component in the main bundle and the component v1 is good to be deprecated in the next major release, team announces the deprecation and the promotion in the [Primer changelog](https://github.com/github/primer/discussions/categories/primer-changelog) (GitHub staff only) under the `Coming soon` section and release a technical preview. Please see an example of the [technical preview announcement for the v35](https://github.com/primer/react/discussions/1918).
```

Comment:
I updated this section to reflect @rezrah's [comments on versioning docs](https://github.com/primer/react/pull/2753/files#r1085554252) ✨